code/modules/vehicles/mecha/mech_melee_attack.dm

```diff
@@ -46,40 +46,46 @@
 		return
 	return ..()
 
-/mob/living/mech_melee_attack(obj/vehicle/sealed/mecha/mecha_attacker, mob/user)
-	if(user.a_intent == INTENT_HARM)
-		if(HAS_TRAIT(user, TRAIT_PACIFISM))
-			to_chat(user, "<span class='warning'>You don't want to harm other living beings!</span>")
-			return
-		mecha_attacker.do_attack_animation(src)
-		if(mecha_attacker.damtype == "brute")
-			step_away(src, mecha_attacker, 15)
-		switch(mecha_attacker.damtype)
-			if(BRUTE)
-				Unconscious(20)
-				take_overall_damage(rand(mecha_attacker.force/2, mecha_attacker.force))
-				playsound(src, 'sound/weapons/punch4.ogg', 50, TRUE)
-			if(BURN)
-				take_overall_damage(0, rand(mecha_attacker.force * 0.5, mecha_attacker.force))
-				playsound(src, 'sound/items/welder.ogg', 50, TRUE)
-			if(TOX)
-				mecha_attacker.mech_toxin_damage(src)
-			else
+/mob/living/mech_melee_attack(obj/vehicle/sealed/mecha/mecha_attacker, mob/living/user)
+	if(isliving(user))
```

Comment:
Early return here? Should diminish the diff as well.

Reply:
I considered that but if you're not in combat mode it also needs to move down to the push section if that makes sense (2 lines below)